090-add-connect-to-test-container.md

```diff
@@ -0,0 +1,153 @@
+     */
+    public void start() { }
+
+    /**
+     * Stop the Kafka Connect cluster.
+     */
+    public void stop() { }
+
+    /**
+     * Return the REST API endpoint of one of the available workers
+     * @return the REST API endpoint
+     */
+    public String getRestEndpoint() { }
+
+    /**
+     * Builder class for {@code StrimziConnectCluster}.
+     * <p>
+     * Use this builder to create instances of {@code StrimziConnectCluster}.
+     * You must at least call {@link #withKafkaCluster(StrimziKafkaCluster)}, and 
+     * {@link #withGroupId(String)} before calling {@link #build()}.
+     * </p>
+     */
+    public static class StrimziConnectClusterBuilder {
+
+        /**
+         * Sets the Kafka cluster the Kafka Connect cluster will use to.
+         * @param kafkaCluster the {@link StrimziKafkaCluster} instance
+         * @return the current instance of {@code StrimziConnectClusterBuilder} for method chaining
+         */
+        public StrimziConnectClusterBuilder withKafkaCluster(StrimziKafkaCluster kafkaCluster) { }
+
+        /**
+         * Sets the number of Kafka Connect workers in the cluster.
+         * If not called, the cluster has a single broker
+         *
+         * @param workersNum the number of Kafka Connect workers
+         * @return the current instance of {@code StrimziConnectClusterBuilder} for method chaining
+         */
+        public StrimziConnectClusterBuilder withNumberOfWorkers(int workersNum) { }
+
+        /**
+         * Adds additional Kafka Connect configuration parameters.
+         * These configurations are applied to all workers in the cluster.
+         *
+         * @param additionalConnectConfiguration a map of additional Kafka Connect configuration options
+         * @return the current instance of {@code StrimziConnectClusterBuilder} for method chaining
+         */
+        public StrimziConnectClusterBuilder withAdditionalConnectConfiguration(Map<String, String> additionalConnectConfiguration) { }
+
+        /**
+         * Specifies the Kafka version to be used for the Connect workers in the cluster.
+         * If not called, the latest Kafka version available from {@link KafkaVersionService} will be used.
+         *
+         * @param kafkaVersion the desired Kafka version for the Connect cluster
+         * @return the current instance of {@code StrimziConnectClusterBuilder} for method chaining
+         */
+        public StrimziConnectClusterBuilder withKafkaVersion(String kafkaVersion) { }
+
+        /**
+         * Whether to include the FileStream connectors.
+         * If not called, the FileStream connectors are enabled.
+         *
+         * @param includeFileConnectors Use false to not include the FileStream connectors
+         * @return the current instance of {@code StrimziConnectClusterBuilder} for method chaining
+         */
+        public StrimziConnectClusterBuilder withIncludeFileConnectors(boolean includeFileConnectors) { }
```

Comment:
```suggestion
        public StrimziConnectClusterBuilder withIncludeFileConnectors() { }
```
No need to have a boolean here I think :). By default `false` and if you call such method you simply assign true to a variable.

Reply:
So what will be output of this? Are we keep the proposed behavior, change it in favor of what Maros propose, or mix it? Guess any other connection will be mounted via volumes?

Reply:
I'm happy with the current proposal. For other connectors, users will need to mount the necessary JARs and use `withAdditionalConnectConfiguration()` to properly set the `plugin.path` configuration accordingly.

Reply:
I think better way is to remove boolean and just name it properly. So `withoutFileConnectors()` for example as default value will be true. That way user can turn it off and load their own File connectors. WDYT?

Reply:
What's the reason for having the file connectors opt out rather than opt in? Given that today you have to actively add the file connectors when running Kafka yourself? Are we expecting most tests to need the file connectors?

Reply:
Out of the built-in connectors these are the easiest to run, as the MirrorMaker connector require 2 Kafka clusters (technically HeartbeatConnector only requires 1 cluster but still it's confusing to use it this way).
I expect the FileStreams connectors to satisfy most of the requirements (I think it should be enough to do the tests I want in metrics-reporter). Then users with more advanced requirements can inject and use their own connectors if needed.